src/content/docs/features/http-client.mdx

```diff
@@ -3,11 +3,125 @@ title: HTTP Client
 description: Access the HTTP client written in Rust.
 ---
 
-import Stub from '@components/Stub.astro';
 import PluginLinks from '@components/PluginLinks.astro';
+import { Tabs, TabItem } from '@astrojs/starlight/components';
+import CommandTabs from '@components/CommandTabs.astro';
 
 <PluginLinks plugin="http" />
 
-<Stub>
-	Based on https://github.com/tauri-apps/plugins-workspace/tree/v2/plugins/http
-</Stub>
+Make HTTP requests with the http plugin.
+
+## Setup
+
+Install the http plugin to get started.
+
+<Tabs>
+    <TabItem label="Automatic">
+
+    1. Use your project's package manager to add the dependency:
+
+    <CommandTabs npm="npm run tauri add http"
+    yarn="yarn run tauri add http"
+    pnpm="pnpm tauri add http"
+    cargo="cargo tauri add http" />
+
+    2. Modify `lib.rs` to initialize the plugin:
+
+    {/* TODO: Revise once https://github.com/tauri-apps/tauri/issues/7696 is in */}
+
+    ```rust ins={6}
+    // lib.rs
+    #[cfg_attr(mobile, tauri::mobile_entry_point)]
+    pub fn run() {
+        tauri::Builder::default()
+            // Initialize the plugin
+            .plugin(tauri_plugin_http::init())
+            .run(tauri::generate_context!())
+            .expect("error while running tauri application");
+    }
+    ```
+
+    </TabItem>
+    <TabItem label="Manual">
+
+    1. Run `cargo add tauri-plugin-http` to add the plugin to the project's dependencies in `Cargo.toml`.
+
+    2. Modify `lib.rs` to initialize the plugin:
+
+    ```rust ins={6}
+    // lib.rs
+    #[cfg_attr(mobile, tauri::mobile_entry_point)]
+    pub fn run() {
+        tauri::Builder::default()
+            // Initialize the plugin
+            .plugin(tauri_plugin_http::init())
+            .run(tauri::generate_context!())
+            .expect("error while running tauri application");
+    }
+    ```
+
+    3. If you'd like to make http requests in JavaScript then install the npm package as well:
+
+    <CommandTabs
+        npm="npm install @tauri-apps/plugin-http"
+        yarn="yarn add @tauri-apps/plugin-http"
+        pnpm="pnpm add @tauri-apps/plugin-http"
+    />
+
+    </TabItem>
+
+</Tabs>
+
+## Usage
+
+The http plugin is available in both as an JavaScript API and in Rust as a [reqwest](https://docs.rs/reqwest/) re-export.
+
+
+### JavaScript
+
+#### 1. Configure allowed scope
```

Comment:
Instead of these being headings, let's keep them as normal text (similar to how we format the steps under the Setup heading).